File: app/src/main/java/com/sortedqueue/programmercreek/fragments/TutorialModelFragment.java
package com.sortedqueue.programmercreek.fragments;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.sortedqueue.programmercreek.R;
import com.sortedqueue.programmercreek.database.TutorialModel;
import com.sortedqueue.programmercreek.interfaces.TutorialNavigationListener;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by Alok on 15/05/17.
 */

public class TutorialModelFragment extends Fragment implements View.OnClickListener {

    @BindView(R.id.titleTextView)
    TextView titleTextView;
    @BindView(R.id.subTitleTextView)
    TextView subTitleTextView;
    @BindView(R.id.slideImageView)
    ImageView slideImageView;
    @BindView(R.id.cancelButton)
    Button cancelButton;
    @BindView(R.id.nextButton)
    Button nextButton;
    @BindView(R.id.slideImageLayout)
    FrameLayout slideImageLayout;
    @BindView(R.id.buttonLayout)
    LinearLayout buttonLayout;
    private TutorialModel tutorialModel;
    private int index = 0;
    private int size;

    private TutorialNavigationListener tutorialNavigationListener;

    public void setParameter(TutorialModel parameter, int index, int size) {
        this.tutorialModel = parameter;
        this.index = index;
        this.size = size;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof TutorialNavigationListener) {
            tutorialNavigationListener = (TutorialNavigationListener) context;
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        tutorialNavigationListener = null;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_tutorial, container, false);
        ButterKnife.bind(this, view);
        titleTextView.setText("Step : " + index);
        subTitleTextView.setText(tutorialModel.getStepDescription());
        slideImageView.setImageDrawable(ContextCompat.getDrawable(getContext(), R.drawable.splash_logo));
        if (index == 1) {
            cancelButton.setVisibility(View.GONE);
            nextButton.setVisibility(View.VISIBLE);
            nextButton.setText("Next");
            cancelButton.setText("Cancel");
        } else if (index > 1 && index < size) {
            cancelButton.setVisibility(View.VISIBLE);
            nextButton.setVisibility(View.VISIBLE);
            nextButton.setText("Next");
            cancelButton.setText("Prev");
        } else if (index == size) {
            cancelButton.setVisibility(View.VISIBLE);
            nextButton.setVisibility(View.VISIBLE);
            nextButton.setText("Done");
            cancelButton.setText("Prev");
        }
        nextButton.setOnClickListener(this);
        cancelButton.setOnClickListener(this);
        if (tutorialModel.getStepImageUrl().trim().length() == 0) {
            slideImageLayout.setVisibility(View.GONE);
        }
        else {
            slideImageLayout.setVisibility(View.VISIBLE);
            Glide.with(getContext())
                    .load(tutorialModel.getStepImageUrl())
                    .asBitmap()
                    .fitCenter()
                    .error(R.color.md_blue_600)
                    .placeholder(R.color.md_blue_600)
                    .into(slideImageView);
        }
        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();

    }

    @Override
    public void onClick(View v) {
        Button button = (Button) v;
        String buttonText = button.getText().toString();
        switch (buttonText) {
            case "Done":
                tutorialNavigationListener.onCancelClick();
                break;
            case "Prev":
                tutorialNavigationListener.onPreviousClick();
                break;
            case "Next":
                tutorialNavigationListener.onNextClick();
                break;
        }
    }
}

File: app/src/main/java/com/sortedqueue/programmercreek/fragments/TutorialModelFragment.kt
package com.sortedqueue.programmercreek.fragments

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

import com.bumptech.glide.Glide
import com.sortedqueue.programmercreek.R
import com.sortedqueue.programmercreek.database.TutorialModel
import com.sortedqueue.programmercreek.interfaces.TutorialNavigationListener
import kotlinx.android.synthetic.main.fragment_tutorial.*


/**
 * Created by Alok on 15/05/17.
 */

class TutorialModelFragment : Fragment(), View.OnClickListener {


    private var tutorialModel: TutorialModel? = null
    private var index = 0
    private var size: Int = 0

    private var tutorialNavigationListener: TutorialNavigationListener? = null

    fun setParameter(parameter: TutorialModel, index: Int, size: Int) {
        this.tutorialModel = parameter
        this.index = index
        this.size = size
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is TutorialNavigationListener) {
            tutorialNavigationListener = context
        }
    }

    override fun onDetach() {
        super.onDetach()
        tutorialNavigationListener = null
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater!!.inflate(R.layout.fragment_tutorial, container, false)


        return view
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        titleTextView!!.text = "Step : " + index
        subTitleTextView!!.text = tutorialModel!!.stepDescription
        slideImageView!!.setImageDrawable(ContextCompat.getDrawable(context, R.drawable.splash_logo))
        if (index == 1) {
            cancelButton!!.visibility = View.GONE
            nextButton!!.visibility = View.VISIBLE
            nextButton!!.text = "Next"
            cancelButton!!.text = "Cancel"
        } else if (index > 1 && index < size) {
            cancelButton!!.visibility = View.VISIBLE
            nextButton!!.visibility = View.VISIBLE
            nextButton!!.text = "Next"
            cancelButton!!.text = "Prev"
        } else if (index == size) {
            cancelButton!!.visibility = View.VISIBLE
            nextButton!!.visibility = View.VISIBLE
            nextButton!!.text = "Done"
            cancelButton!!.text = "Prev"
        }
        nextButton!!.setOnClickListener(this)
        cancelButton!!.setOnClickListener(this)
        if (tutorialModel!!.stepImageUrl.trim { it <= ' ' }.length == 0) {
            slideImageLayout!!.visibility = View.GONE
        } else {
            slideImageLayout!!.visibility = View.VISIBLE
            Glide.with(context)
                    .load(tutorialModel!!.stepImageUrl)
                    .into(slideImageView!!)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()

    }

    override fun onClick(v: View) {
        val button = v as Button
        val buttonText = button.text.toString()
        when (buttonText) {
            "Done" -> tutorialNavigationListener!!.onCancelClick()
            "Prev" -> tutorialNavigationListener!!.onPreviousClick()
            "Next" -> tutorialNavigationListener!!.onNextClick()
        }
    }
}
